02-peer-protocol.md

```diff
@@ -49,11 +68,502 @@ transaction is confirmed are also not persistent - until you know the script
 pubkey corresponding to the funding output nothing prevents duplicative channel
 ids.
 
+### `channel_id`, v2
 
-## Channel Establishment
+For channels established using the v2 protocol, the `channel_id` is the
+`SHA256(lesser-revocation-basepoint || greater-revocation-basepoint)`,
+where the lesser and greater is based off the order of the basepoint.
+
+When sending `open_channel2`, the peer's revocation basepoint is unknown.
+A `temporary_channel_id` must be computed by using a zeroed out basepoint
+for the non-initiator.
+
+When sending `accept_channel2`, the `temporary_channel_id` from `open_channel2`
+must be used, to allow the initiator to match the response to its request.
+
+#### Rationale
+
+The revocation basepoints must be remembered by both peers for correct
+operation anyway. They're known after the first exchange of messages,
+obviating the need for a `temporary_channel_id` in subsequent messages.
+By mixing information from both sides, they avoid `channel_id` collisions,
+and they remove the dependency on the funding txid.
```

Comment:
An on-chain transaction ID binds to the following serialization format:

`[nVersion][txins][txouts][nLockTime]`

For legacy channels (v1 funding flow), the `txins`, the `txouts` (apart of the funding output) and the `nLockTime` (for the value in the past) constitute an information space that can be freely tweaked by the funding counterparty. This information space tweaking can be iterated until a collision for a known v2 channel id is found (same process than finding proof-of-work for miner). Once a collision is found this id could be announced through the v1 funding flow and provoke “undefined behavior” in the counterparty’s LN implementation (e.g force-close of the colliding channels).

Of course, this concern is not practical as SHA256 has preimage resistance as a hash function. Still, if more channels ids are introduced in the future, implementations could recommend to prefix their channel id contexts. 
This could be of interest today, as BOLT3 does not have recommendation on `revocation_basepoint` derivation, and multi-party curve point construction (e.g MuSig2) opens the door to all the attack vectors affecting distributed key generation.